src/api/hydrateCharge.ts

```diff
@@ -0,0 +1,41 @@
+import { PAY_HYDRATE_CHARGE } from '../network/definitions/pay';
+import { sendRequest } from '../network/request';
+import { UNCAUGHT_HYDRATE_CHARGE_ERROR_CODE } from '../pay/constants';
+import { getPayErrorCode } from '../pay/utils/getPayErrorCode';
+import type {
+  HydrateChargeAPIParams,
+  HydrateChargeParams,
+  HydrateChargeResponse,
+} from './types';
+
+export async function hydrateCharge({
+  sender,
+  chargeId,
+}: HydrateChargeParams): Promise<HydrateChargeResponse> {
+  try {
+    const res = await sendRequest<
+      HydrateChargeAPIParams,
+      HydrateChargeResponse
+    >(PAY_HYDRATE_CHARGE, [
+      {
+        sender: sender,
+        chainId: 8453,
```

Comment:
for Chain ID, let's use viem.

so it is more explicity like `base.id`

Also, we probably want to get the chain from the config.

So even better this API should have a `chainId` as argument.